### ISSUE TITLE
Allow global*($prop) usage with agg references (clusterAvg($prop))

### ISSUE THREAD

**Issue by davidmanzanares** [CartoDB/carto-vl#407]
We should enable the usage of `global*($property)` expressions even when the same property is referenced inside a cluster aggregation function.

**Comment by davidmanzanares** [comment]
Well... It works, but... I think it's 99% luck. Moreover, there are some possible optimizations around this problem.

Right now we are not controlling in CARTO VL this case at all. In fact, this is what we send to Maps API:
```
{
    "buffersize": {
        "mvt": 1
    },
    "layers": [
        {
            "type": "mapnik",
            "options": {
                "sql": "SELECT count,cartodb_id,the_geom,the_geom_webmercator FROM traffic_accidents_copy ",
                "aggregation": {
                    "columns": {
                        "_cdb_agg_avg_count": {
                            "aggregate_function": "avg",
                            "aggregated_column": "count"
                        }
                    },
                    "dimensions": {
                        "count": "count"
                    },
                    "placement": "centroid",
                    "resolution": 1,
                    "threshold": 1
                }
            }
        }
    ]
}
```

Maps API instantiation doesn't fail, although it should (right?) fail.

It would be easy to change CARTO VL to stop asking for globally aggregated data, but, in that case, we would need to get the metadata for all the columns in all cases (this may or may not hurt performance significantly in GeoJSON).

Another option would be to have two lists of columns: data required ones and metadata required ones. However, this would require more work.

cc: @jgoizueta

**Comment by davidmanzanares** [comment]
Regarding the new point of the lack of control of the situation with un-aggregated and aggregated properties, this was already fixed (I tested it but I haven't search for a PR id).

Regarding the fix for this, we could abuse the MNS internal API more to hack more with strings. However, since @jgoizueta is going to work in temporal aggs., it doesn't make sense to leave this internal API.

My plan is to propose a new internal API for the MNS, get @jgoizueta review on it, apply it on existing expressions fixing this along the way and paving the way for the temporal agg expressions.

**Comment by davidmanzanares** [comment]
Proposal for the MNS type:

```
const mns = new Set([
  {
  name: 'price',
  aggregations: new Set(['avg', 'max])
  usage: 'all'
  },
  {
  name: 'amount',
  usage: 'all'
  },
  {
  name: 'other',
  usage: 'metadata'
  },
  {
  name: 'date',
  granurality: ??? // to be defined later by @jgoizueta
  usage: 'all'
  }
]);
```

**Comment by davidmanzanares** [comment]
Since the proposed format had many problems we switched to https://github.com/CartoDB/carto-vl/pull/893/files#diff-891c2cbdeca14d1f0c6d58b998f4f86bR5

**Comment by davidmanzanares** [comment]
Fixed by #893
